pwndbg/gdblib/heap/jemalloc.py

```diff
@@ -0,0 +1,488 @@
+from __future__ import annotations
+
+import gdb
+
+import pwndbg.gdblib.info
+import pwndbg.gdblib.memory
+import pwndbg.gdblib.typeinfo
+
+RTREE_HEIGHT = 2
+# adapted from jemalloc source 5.3.0
+LG_VADDR = 48
+LG_PAGE = 12
+RTREE_NLIB = LG_PAGE
+# jemalloc/include/jemalloc/internal/jemalloc_internal_types.h
+MALLOCX_ARENA_BITS = 12
+# obj/include/jemalloc/jemalloc.h
+LG_SIZEOF_PTR = 3
+
+RTREE_NSB = LG_VADDR - RTREE_NLIB
+RTREE_NHIB = (1 << (LG_SIZEOF_PTR + 3)) - LG_VADDR
+
+
+# TODO: Move to relevant place
+# include/jemalloc/internal/edata.h
+
+
+def mask(current_field_width, current_field_shift):
+    return ((1 << current_field_width) - 1) << current_field_shift
+
+
+LG_QUANTUM = 4  # TODO: lookup value acc to architecture from include/jemalloc/internal/quantum.h (currently set for arch64)
```

Comment:
The arch name is 'aarch64' or 'arm64'. But anyway, can we make this work for x86-64 as well?

Reply:
According to https://github.com/jemalloc/jemalloc/blob/5.3.0/include/jemalloc/internal/quantum.h#L21, value for it is 4 as well so we can remove this TODO.